sdk/rust-examples/logistic-regression/src/main.rs

```diff
@@ -172,14 +171,14 @@ fn flatten(ds: &[Dataset]) -> Dataset {
 /// `return_code::ErrorCode::BadInput` if the bytes cannot be decoded from
 /// `pinecone` into a `Dataset` value, or if the dataset is not self-consistent
 /// with respect to the dimensions of each of its points.
-fn read_dataset(input: &[u8]) -> Result<Dataset, ErrNo> {
+fn read_dataset(input: &[u8]) -> Result<Dataset, i32> {
     match pinecone::from_bytes::<Dataset>(input) {
-        Err(_err) => Err(ErrNo::Proto),
+        Err(_err) => Err(-1),
         Ok(data) => {
             if data.self_consistent() {
                 Ok(data)
             } else {
-                Err(ErrNo::Proto)
+                Err(-1)
```

Comment:
Just something to note, most applications use positive error codes because most OSes truncate the error code to a u8:

https://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html

>    As explained in other sections, certain exit status values have been reserved for special uses and should be used by applications only for those purposes:
>```
>     126
>        A file to be executed was found, but it was not an executable utility.
>     127
>        A utility to be executed was not found.
>     >128
>        A command was interrupted by a signal.
>```
>    The behavior of exit when given an invalid argument or unknown option is unspecified, because of differing practices in the various historical implementations. A value larger than 255 might be truncated by the shell, and be unavailable even to a parent process that uses waitid() to get the full exit value. It is recommended that implementations that detect any usage error should cause a non-zero exit status (or, if the shell is interactive and the error does not cause the shell to abort, store a non-zero value in "$?" ), but even this was not done historically in all shells.

But I don't think strict compatibility with Linux processes is a goal. Returning non-truncated i32s is probably more useful.

Actually, thinking about it more, it may be possible to use Rust's extra `main` sugar in the examples. But maybe this should be addressed in another PR.

https://www.dev-notes.eu/2020/03/Return-a-Result-Type-from-the-Main-Function-in-Rust/

Reply:
I will flip the error code to `1`. I just release WASI `proc_exit` take in `u32` rather than `i32`. Will also update it.
I agree some of any further functionality should be included in a separate PR. For example, the `main` sugar you mentioned. For more information, we could provide customer Terminator https://docs.rs/terminator/0.1.0/terminator/ which enable the sugar you mentioned.

Reply:
Ah, so it does.

The extra sugar could be looked at more when addressing debugability:
https://github.com/veracruz-project/veracruz/issues/155